flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -669,6 +669,13 @@ public void setNumBuffers(int numBuffers) {
 
             currentPoolSize = Math.min(numBuffers, maxNumberOfMemorySegments);
 
+            // reset overdraft buffers
+            while (numberOfRequestedOverdraftMemorySegments > 0
+                    && numberOfRequestedMemorySegments < currentPoolSize) {
```

Comment:
For the case that the pool size becomes smaller, I prefer not to do special handling, because we already allow the case that `numRequestedBuffers` exceed the `pool size` after `setNumbers`. WDYT?

Reply:
I see it can work now due to we return overdraft buffer first, right?

However I think convert `numberOfRequestedMemorySegments` to `numberOfRequestedOverdraftMemorySegments ` is more clear, and the semantic of these fields are more explicit.

I'm not sure which one is better. WDYT?

Reply:
The reason why I didn't do this convert here is that we have an upper limit on the number of overdraft buffers. If this conversion is allowed, and the `pool size` changes very small, the `numberOfRequestedOverdraftMemorySegments ` will exceed its upper limit.
Of course, we can convert it to the upper limit at most, but this will make the logic a little complicated. It seems that there is no benefit to introduce this mechanism. WDTY?